src/snowflake/snowpark/_internal/utils.py

```diff
@@ -1175,17 +1175,6 @@ def get_temp_type_for_object(use_scoped_temp_objects: bool, is_generated: bool)
     )
 
 
-def check_is_pandas_dataframe_in_to_pandas(result: Any) -> None:
-    if not isinstance(result, pandas.DataFrame):
-        raise SnowparkClientExceptionMessages.SERVER_FAILED_FETCH_PANDAS(
-            "to_pandas() did not return a pandas DataFrame. "
-            "If you use session.sql(...).to_pandas(), the input query can only be a "
-            "SELECT statement. Or you can use session.sql(...).collect() to get a "
-            "list of Row objects for a non-SELECT statement, then convert it to a "
-            "pandas DataFrame."
-        )
-
-
```

Comment:
can we understand why there was such a check at the very beginning?
I'm thinking removing this check would have unexpected side effect

Reply:
this check is meant to make sure user do not call to_pandas on non-select statements.
As for reason, I think @sfc-gh-jdu  can answer